code/modules/mob/living/basic/pets/parrot/parrot.dm

```diff
@@ -0,0 +1,406 @@
+GLOBAL_LIST_INIT(strippable_parrot_items, create_strippable_list(list(
+	/datum/strippable_item/parrot_headset,
+)))
+
+
+/// Parrots! Klepto bastards that imitate your speech and hoard your shit.
+/mob/living/basic/parrot
+	name = "parrot"
+	desc = "The parrot squawks, \"They're a Parrot! BAWWK!\""
+	icon = 'icons/mob/simple/animal.dmi'
+	icon_state = "parrot_fly"
+	icon_living = "parrot_fly"
+	icon_dead = "parrot_dead"
+	density = FALSE
+	health = 80
+	maxHealth = 80
+	pass_flags = PASSTABLE | PASSMOB
+
+	butcher_results = list(/obj/item/food/cracker = 1)
+	melee_damage_upper = 10
+	melee_damage_lower = 5
+
+	response_help_continuous = "pets"
+	response_help_simple = "pet"
+	response_disarm_continuous = "gently moves aside"
+	response_disarm_simple = "gently move aside"
+	response_harm_continuous = "swats"
+	response_harm_simple = "swat"
+	combat_mode = TRUE //parrots now start "aggressive" since only player parrots will nuzzle.
+	attack_verb_continuous = "chomps"
+	attack_verb_simple = "chomp"
+	attack_vis_effect = ATTACK_EFFECT_BITE
+	friendly_verb_continuous = "grooms"
+	friendly_verb_simple = "groom"
+	mob_size = MOB_SIZE_SMALL
+	gold_core_spawnable = FRIENDLY_SPAWN
+
+	ai_controller = /datum/ai_controller/basic_controller/parrot
+
+	/// Icon we use while sitting
+	var/icon_sit = "parrot_sit"
+	/// The number of damage we do when we decide to aggro for our lives
+	var/parrot_damage_upper = 10
+	/// Potential bodyparts for us to attack
+	var/parrot_dam_zone = CARBON_GENERIC_BODY_ZONES
+
+	///Headset for Poly to yell at engineers :)
+	var/obj/item/radio/headset/ears = null
+
+	///Parrots are kleptomaniacs. This variable ... stores the item a parrot is holding.
+	var/obj/item/held_item = null
+
+	/// The blackboard key we use to store the string we're repeating
+	var/speech_blackboard_key = BB_PARROT_REPEAT_STRING
+	/// The generic probability odds we have to do a speech-related action // FIXME might need to tone this down
+	var/speech_probability_rate = 5
+	/// The generic probability odds we have to switch out our speech string
+	var/speech_shuffle_rate = 30
+
+	//Parrots will generally sit on their perch unless something catches their eye.
+	var/static/list/desired_perches = list(
```

Comment:
should make this a typecache since find_and_set/in_list now works with typecaches